### ISSUE TITLE
Unable to login as admin on docker

### ISSUE THREAD

**Issue by mekarpeles** [internetarchive/openlibrary#1197]
Multiple users have reported problems logging in as the dev admin user `openlibrary` and instead become logged in as `openlibrary7774` or some other random number appended.

It looks like the cause is getting the admin123 password (an infogami default) incorrect the first time locks subsequent logins to the same non-privileged user `openlibrary{random-number}`

Clearing browser cookies and logging in again with the correct password works around this issue.

Not sure why the login uses the stored user in the cookie instead of what was entered by the user logging in -- this seems like a bug and should be investigated / fixed.

**ACTION:**  For local dev login, stop old saved cookies from overriding the currently entered username the dev is trying to login with.

**Comment by hornc** [comment]
I can login to the docker env as openlibrary@example.com, pw: admin123 which is the test user

**Comment by mekarpeles** [comment]
Can you access any of the admin routes?

**Comment by hornc** [comment]
Turns out no, I can't, I get an error:
`/openlibrary/openlibrary/templates/admin/index.html: error in processing template: AttributeError: 'reading_log' (falling back to default template)`

I obviously haven't used the admin endpoint for a while, but believed (and still do) that the user's admin access is ok because I was importing and editing with admin privs.

**Comment by hornc** [comment]
Error coming from here: https://github.com/internetarchive/openlibrary/blob/master/openlibrary/templates/admin/index.html#L172
```
/openlibrary/openlibrary/templates/admin/index.html in __template__
172.    <td class="amount">$commify(counts.reading_log['total_books_logged']['week']['count'])</td> 
```

**Comment by hornc** [comment]
@mekarpeles .... and unfortunately I see the same thing in production when going to https://openlibrary.org/admin?debug=true

**Comment by hornc** [comment]
re. the original question, I can access pages off `/admin`, eg http://localhost:8080/admin/people it's only the main admin page that gives the error.

**Comment by mekarpeles** [comment]
/admin fixed on dev.openlibrary.org (but is separate issue). It moved to /stats

There are other admin endpoints e.g. /admin/people which don't work for me when I login via docker or vagrant

**Comment by hornc** [comment]
@mekarpeles I have tried the docker instructions from a fresh checkout of this repo and clean docker builds a+ volumes, and can log in as openlibrary@example.com and access the admin pages.
`docker-compose up`

**Comment by jdlrobson** [comment]
@tfmorris I'm still hitting this issue. I can't login as an admin. Every time I log in I end up as the user openlibrary7774

I for some reason have the user "openlibrary7774" locally. @hornc pointed out this could be a file permission issue.

**Comment by tfmorris** [comment]
I am able to successfully log in as an admin and see the following

* admin - redirects to stats
* people - renders
* loans - renders
* waitinglists - relation "waitingloan" does not exist LINE 1: SELECT * FROM waitingloan WHERE status='available' ORDER BY ... ^
    Python | /usr/local/lib/python2.7/dist-packages/DBUtils/SteadyDB.py in tough_method, line 604
* block - renders
* spamwords - renders
* solr - renders
* imports -  relation "import_item" does not exist LINE 1: SELECT count(*) as count FROM import_item WHERE status='pend... ^ 
    Python | /usr/local/lib/python2.7/dist-packages/DBUtils/SteadyDB.py in tough_method, line 604
* graphs - broken images
* inspect store - renders
* inspect memcache - renders

In summary it mostly works except for two errors which look like they could be because of missing data.

@mekarpeles @hornc Are you able to log in as admin now?

**Comment by jdlrobson** [comment]
For some reason when i login as any email address i end up assigned the username openlibrary1083.

When i try to register a new account i get email already registered...

**Comment by jdlrobson** [comment]
Is there a way i can delete all users in database to see if user creation is associated with this problem?

**Comment by mekarpeles** [comment]
I'm hitting the same issue as @jdlrobson and we're hearing corroboration from @rohitjain00 who is also hitting this issue

**Comment by shreyansh23** [comment]
@mekarpeles 
> I'm hitting the same issue as @jdlrobson and we're hearing corroboration from @rohitjain00 who is also hitting this issue

Yes, the issue is still there.

**Comment by mekarpeles** [comment]
The admin user is defined within infogami repo in `infogami/utils/delegate.py` within `admin_login` and the `admin_password` is specified within `openlibrary/config/openlibrary.yml` within this repo.

The hypothesis is, use attempts to login with an email. The email and password don't match. The user `openlibrary` is deemed to already exist. Numbers are added onto the username. The username is created and the user is able to login but no longer as the admin user.

**Comment by mekarpeles** [comment]
@hornc can you please update this task w/ the cache clearing fix?

**Comment by hornc** [comment]
Here are the steps to clear cookies and re-enable admin login if you get stuck with a non-admin user:

1) `docker-compose down`
2)  completely clear all localhost / 0.0.0.0 / 127.0.0.1 cookies from your browser
3)  `docker-compose up`
4) then login with `openlibrary@example.com` pw: `admin123`   (It's possible there are acceptible variations on those credentials, but those should definitely work)

**Comment by tabshaikh** [comment]
https://github.com/internetarchive/openlibrary/issues/1197#issuecomment-479752932 @charles should we add this in the FAQ too

**Comment by hornc** [comment]
@tabshaikh Good idea, but I think there is a bug that needs to be fixed first -- I have added a description to clarify the action I think needs to be taken to make the fix. Hopefully that will make the workaround unnecessary, but it needs some investigation and testing.

**Comment by mekarpeles** [comment]
Closing this issue as resolved. Updated the #2049 guide to reflect. We should probably open a new issue for solving the root of the problem (which requires better documentation) -- 

Fix login on dev instance to fail if incorrect admin credentials
If user on dev enters wrong admin email + password and spoofs login, unable to login as admin. The flow should be fixed

**Comment by cdrini** [comment]
See also: https://github.com/internetarchive/openlibrary/issues/2050#issuecomment-830569331
